### ISSUE TITLE
Converting form encoded parameters to an object

### ISSUE THREAD

**Issue by KarboniteKream** [line/armeria#3577]
I was working on an annotated service method to handle `POST application/x-www-form-urlencoded`, which means the body of the request looks like `foo=1&bar=2`.

My initial idea was to define a request object and annotated method like this:
```java
public class MyRequest {
    String foo;
    String bar;
}

@Post("/")
public String myPost(@RequestObject MyRequest request) {
    return request.foo;
}
```

But it looks like Armeria only deserializes the request like this if the specified content type is JSON, so the solution was to use `QueryParams` or annotate `MyRequest` fields with `@Param`.

I was wondering if Armeria could add support for converting the parameters in this case as well or not. Perhaps the use case is very low (even in our case it's only one endpoint, which we'll most likely replace), so this might be feature creep. What do you think?

**Comment by trustin** [comment]
I agree with you but we need to define something that's not confusing to our users. What do you think about introducing an annotation which adds a `RequestConverterFunction`? e.g.

```java
@Post("/query")
@Form // Adds FormRequestConverterFunction
public String post(MyRequest request) {
  ...
}

@Get("/get")
@Form // Could also work for GET requests, sourcing from the query string in path
public ... { ... }
```

How do we map the query parameters into MyRequest? I think it'd be awesome if we can use Jackson for this. For example, it'd be nice if we can tell Jackson to create `MyRequest, just like it received a JSON object. For example, if query string is `foo=bar&alice=bob`, we could feed Jackson with a `JsonNode` that represents `{ "foo": "bar", "alice": "bob" }`. WDYT?
